### PR TITLE
fix: `doRequestFollowRedirectsBuffer` cannot get body in HTTP2 scenario

### PR DESCRIPTION
**Pull request by Duslia**
#### What type of PR is this?

fix
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
在 HTTP2 场景下，`doRequestFollowRedirectsBuffer` 不能获取到 body

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: `doRequestFollowRedirectsBuffer` cannot get body in HTTP2 scenario. The affected interfaces include  `client.Get`，`client.GetTimeout`，`client.GetDeadLine`，`client.Post`
zh(optional): 在 HTTP2 场景下，`doRequestFollowRedirectsBuffer` 不能获取到 body，受影响的接口包括 `client.Get`，`client.GetTimeout`，`client.GetDeadLine`，`client.Post`

### CODE REVIEW

**Review by li-jin-gou — DISMISSED** [review]

File: pkg/app/client/client_test.go [hunk 1]
```diff
@@ -165,7 +166,7 @@ func TestClientGetWithBody(t *testing.T) {
 	opt.Addr = "unix-test-10002"
 	opt.Network = "unix"
 	engine := route.NewEngine(opt)
-	engine.GET("/", func(c context.Context, ctx *app.RequestContext) {
+	engine.POST("/", func(c context.Context, ctx *app.RequestContext) {
```
Comment (welkeyever) [inline pkg/app/client/client_test.go:169]: Why change this line?

File: pkg/protocol/client/client.go [hunk 2]
```diff
@@ -220,7 +220,8 @@ func doRequestFollowRedirectsBuffer(ctx context.Context, req *protocol.Request,
 
 	statusCode, _, err = DoRequestFollowRedirects(ctx, req, resp, url, defaultMaxRedirectsCount, c)
 
-	body = bodyBuf.B
+	// In HTTP2 scenario, client use stream mode to create a request and its body is in body stream.
```
Comment (welkeyever) [inline pkg/protocol/client/client.go:223]: Why stress “HTTP2 scenario”? If it is about body stream, HTTP/1.1 also has the kind of issue?

Comment (Duslia) [reply pkg/protocol/client/client.go:223]: HTTP2 is a scenario that must trigger this bug. And in HTTP1, only client recv body exceed max size and client is in stream mode can trig it.

File: pkg/app/client/client_test.go (hunk 1 above)
Comment (welkeyever) [reply pkg/app/client/client_test.go:169]: This func name is `TestClientGetWithBody`?
Maybe add a new test func for the added logic?

**Review by li-jin-gou — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**